src/NSubstitute/Core/CallBaseConfiguration.cs

```diff
@@ -6,6 +6,7 @@ namespace NSubstitute.Core
     public class CallBaseConfiguration : ICallBaseConfiguration
     {
         private ConcurrentQueue<ICallSpecification> Exclusions { get; } = new ConcurrentQueue<ICallSpecification>();
+        private ConcurrentQueue<ICallSpecification> Inclusions { get; } = new ConcurrentQueue<ICallSpecification>();
```

Comment:
Not sure if this is a bad suggestion, so please ignore it if it is... 😄 
What if we use a single collection of `ConcurrentStack<Tuple<ICallSpecification, State>>` where `enum State { Include, Exclude }` or similar? Then we push everything on to that single stack, then search for the most recent match. That would let us override previous settings by pushing a new value (similar to how we can override `.Returns()` values).

Haven't thought through the problems with this yet, but thought I'd mention it as you can probably rule it out faster than I can. 😂

Reply:
Thanks for the brilliant idea! Indeed, it's much better to rework it in a way "last rule wins", to be compliant with other places where we follow the similar approach.

I reworked to be compliant.